src/app/windowIcon.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * @copyright Copyright (c) 2023 Grigorii Shartsev <grigorii.shartsev@nextcloud.com>
+ *
+ * @author Grigorii Shartsev <grigorii.shartsev@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const path = require('node:path')
+const {
+	isLinux,
+	isWindows,
+	isMac,
+} = require('../shared/os.utils.js')
+
+let windowIcon
+
+if (isWindows()) {
+	windowIcon = path.join(__dirname, '../img/icons/icon.ico')
+} else if (isMac()) {
+	windowIcon = path.join(__dirname, '../img/icons/icon.icns')
+} else if (isLinux()) {
+	windowIcon = path.join(__dirname, '../img/icon.png')
+} else {
+	// Also use PNG as default
+	windowIcon = path.join(__dirname, '../img/icon.png')
```

Comment:
```suggestion
	windowIcon = path.join(__dirname, '../img/icons/icon.png')
```

According to [docs](https://www.electronforge.io/guides/create-and-add-icons#configuring-installer-icons), this also should be added to forge.config.js:
```js
      name: '@electron-forge/maker-deb', //same for zip, i guess
        config: {
          options: {
            icon: './img/icons/icon.png'
          }
      }
```

But it still doesn't appear on Linux Dock

Reply:
> According to docs, this also should be added to forge.config.js:

Yes, but we don't plan to build distributable before final release

> //same for zip, i guess

No, zip is just an archive, it doesn't have any icons or OS specific "package" settings.